internal/pkg/enricher/main.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package enricher
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"regexp"
+	"syscall"
+
+	"github.com/cirocosta/dmesg_exporter/kmsg"
+	"github.com/cirocosta/dmesg_exporter/reader"
+	"github.com/go-logr/logr"
+)
+
+var (
+	logFile = "/dev/kmsg"
+	logger  logr.Logger
+)
+
+// Run the log-enricher to scrap audit logs and enrich them with
+// Kubernetes data (namespace, pod and container).
+func Run(l logr.Logger) {
+	logger = l
+	nodeName := os.Getenv("NODE_NAME")
+	if nodeName == "" {
+		logger.Error(nil, "'NODE_NAME' environment variable not found")
+		os.Exit(1)
+	}
+
+	logger.V(1).Info("starting log-exporter on node: ", nodeName)
+
+	auditLines := make(chan string)
+	go tailDevice(logFile, auditLines)
+
+	for {
+		line := <-auditLines
+
+		if !isAuditLine(line) {
+			continue
+		}
+
+		auditLine, err := extractAuditLine(line)
+		if err != nil {
+			fmt.Printf("extract seccomp details from audit line: %v\n", err)
+		}
+
+		if auditLine.SystemCallID == 0 {
+			continue
+		}
+
+		cID := getContainerID(auditLine.ProcessID)
+		containers, err := getNodeContainers(nodeName)
+		c, found := containers[cID]
+
+		if !found {
+			logger.Error(err, "containerID not found", "processID", auditLine.ProcessID)
+			continue
+		}
+
+		name := systemCalls[auditLine.SystemCallID]
+		fmt.Printf("audit(%s) type=%s node=%s pid=%d ns=%s pod=%s c=%s exe=%s syscall=%s\n",
+			auditLine.TimestampID, auditLine.Type, nodeName,
+			auditLine.ProcessID, c.Namespace, c.PodName,
+			c.ContainerName, auditLine.Executable, name)
+	}
+}
+
+func blockAndCancelOnSignal(cancel context.CancelFunc) {
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGTERM, syscall.SIGINT)
+	<-sigChan
+
+	cancel()
+}
+
+func tailDevice(device string, msgs chan string) {
+	file, err := os.Open(device)
+	if err != nil {
+		return
+	}
+	defer file.Close()
+
+	// seek to the end of device
+	_, err = file.Seek(0, os.SEEK_END)
```

Comment:
`os.SEEK_END` appears to be deprecated: https://golang.org/pkg/os/#pkg-constants

Reply:
Great point, I replaced that with `io.SeekEnd` instead.